pkg/openai/client.go

```diff
@@ -446,6 +455,22 @@ func (c *Client) Call(ctx context.Context, messageRequest types.CompletionReques
 	return &result, nil
 }
 
+func isO1Model(model string, envs []string) bool {
```

Comment:
there is two ways to check whether the model is used by o1

1. check if modeName is o1. This is true if used with standalone gptscript.
2. Check if `OPENAI_MODEL_NAME` is set. This will be set in Obot to determine the name.